alf/utils/dist_utils.py

```diff
@@ -434,6 +430,40 @@ def greedy_fn(dist):
         return nest.map_structure(greedy_fn, nested_distributions)
 
 
+def get_mode(dist):
+    """Get the mode of the distribution. Note that if dist is a transformed
+        distribution, the result may not be the actual mode of dist.
+
+    Args:
+        dist (td.Distribution)
+    Returns:
+        The mode of the distribution. If dist is a transformed distribution,
+        the result is calculated by transforming the mode of its base
+        distribution and may not be the actual mode for dist.
+    Raises:
+        NotImplementedError if dist or its base distribution is not
+            td.Categorical, td.Normal, td.Independent or
+            td.TransformedDistribution.
+    """
+    if isinstance(dist, td.categorical.Categorical):
+        mode = torch.argmax(dist.logits, -1)
+    elif isinstance(dist, td.normal.Normal):
+        mode = dist.mean
+    elif isinstance(dist, td.Independent):
```

Comment:
Does this also support several independent Categorical distributions?

Reply:
Yes, it does. Added a test case on this.